### PR TITLE
Indexes a Content.ContentItem.Parent field in Lucene for ContainedPart.

### LINKED ISSUE #3794: select blogposts liquid query
how can i get blogpost for one blog? i have 3 blogs and i want to get post of any of them
{
"query": {
"term" : { "Content.ContentItem.ContentType" : "BlogPost" }
},
"sort": { "Content.ContentItem.CreatedUtc": { "order": "desc" } },
"size": 2
}

how i should set my condition? i can see any value related to blog code. maybe we need to access parent of the list in liquid.

### PR DESCRIPTION
**Pull request by Skrypt**
Fixes #3794, #3949

Need to filter to see if it will fix also other issues that I missed.

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Lists/Indexes/ListPartContentIndexHandler.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+using System.Threading.Tasks;
+using OrchardCore.ContentManagement;
+using OrchardCore.Indexing;
+using OrchardCore.Lists.Models;
+
+namespace OrchardCore.Lists.Indexes
+{
+    public class ListPartContentIndexHandler : IContentItemIndexHandler
```
Comment (sebastienros) [inline src/OrchardCore.Modules/OrchardCore.Lists/Indexes/ListPartContentIndexHandler.cs]: Why isn't it a `ContentPartIndexHandler<ContainedPart>` then ?

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Lists/Indexes/ListPartContentIndexHandler.cs]: True and easier 😄

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Lists/Indexes/ListPartContentIndexHandler.cs]: Strange but it's not working now with these changes.

Comment (Skrypt) [reply src/OrchardCore.Modules/OrchardCore.Lists/Indexes/ListPartContentIndexHandler.cs]: Reverting the changes to my previous solution. A ContainedPart is a dynamic welded part which can't be listed from the ContentDefinitionManager.

**Pull request merged**